src/answers/StopwatchAnswerComponent.coffee

```diff
@@ -0,0 +1,87 @@
+React = require 'react'
+H = React.DOM
+R = React.createElement
+
+now = () -> new Date().getTime()
+
+module.exports = class StopwatchAnswerComponent extends React.Component
+  @propTypes:
+    onValueChange: React.PropTypes.func.isRequired
+
+  constructor: (props) ->
+    super
+    @state =
+      timerId: 0       # ID of the running JS timer
+      elapsedTicks: 0  # Tick count
+      editing: false   # True if in edit mode
+      editTicks: 0     # Temporary store for manual-mode tick count
+
+  # Starts a timer to update @elapsedTicks every 10 ms
+  handleStartClick: () =>
+    startTime = now() - @state.elapsedTicks # for restarts we need to fudge the startTime
+    update = () => @setState(elapsedTicks: now() - startTime)
+    @setState(timerId: setInterval(update, 10)) # create a timer and store its id in state
```

Comment:
@grassick I had it that way initially: https://github.com/mWater/mwater-forms/commit/be4f931858f0cddb17f465840004c6c08cc618ec?diff=split.  The problem is that you've got to then store both `startTime` and `stopTime` as state variables, and run `forceUpdate` in a timer anyway to update the display, plus it convolutes logic a bit when you're displaying: you have to display `now()-startTime` when running and `stopTime-startTime` otherwise.  *Then* in manual edit mode you've got one option represented as a start/stop time, and one state represented as ticks, so just an extra source of potential bugs.

So, that's why I thought storing `elapsedTicks` as state seemed like the best option.